api/types/stacks/stacks.go

```diff
@@ -0,0 +1,91 @@
+package stacks // import "github.com/docker/docker/api/types/stacks
+
+import (
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/swarm"
+)
+
+// Stack is an object representing a server-side stack
+type Stack struct {
+	// ID is the ID of the stack, generated by the engine
+	ID string
+	swarm.Meta
+
+	// Spec is the StackSpec defining what this stack should look like
+	Spec StackSpec `json:",omitempty"`
+}
+
+// StackSpec defines the spec for a server-side stack.
+type StackSpec struct {
```

Comment:
Wondering where `Volumes` are kept that are defined as part of the stack; also if the structure should be more flexible somehow 🤔

Reply:
Forgot to define Volumes. But also, how do you mean, "more flexible"?